eth/filters/api.go

```diff
@@ -229,6 +232,80 @@ func (api *PublicFilterAPI) NewHeads(ctx context.Context) (*rpc.Subscription, er
 	return rpcSub, nil
 }
 
+func (api *PublicFilterAPI) NewReturnDataFilter() rpc.ID {
+	var (
+		retCh  = make(chan *types.ReturnData)
+		retSub = api.events.SubscribeReturnData(retCh)
+	)
+
+	api.filtersMu.Lock()
+	api.filters[retSub.ID] = &filter{typ: ReturnDataSubscription, deadline: time.NewTimer(deadline), retdata: make([]types.ReturnData, 0), s: retSub}
+	api.filtersMu.Unlock()
+
+	go func() {
+		for {
+			select {
+			case retdata := <-retCh:
+				api.filtersMu.Lock()
+				if f, found := api.filters[retSub.ID]; found {
+					f.retdata = append(f.retdata, *retdata)
+				}
+				api.filtersMu.Unlock()
+			case <-retSub.Err():
+				api.filtersMu.Lock()
+				delete(api.filters, retSub.ID)
+				api.filtersMu.Unlock()
+				return
+			}
+		}
+	}()
+
+	return retSub.ID
+}
+
+func (api *PublicFilterAPI) ReturnData(ctx context.Context) (*rpc.Subscription, error) {
+	notifier, supported := rpc.NotifierFromContext(ctx)
+	if !supported {
+		return &rpc.Subscription{}, rpc.ErrNotificationsUnsupported
+	}
+
+	txListen := make(map[common.Hash]bool)
```

Comment:
Sounds like a reasonable precaution--I'll take this and your struct{}{} suggestion.

Reply:
I changed my mind on this.  There is already a limit on the number of pending and queued transactions which can be in the txpool.  If the client submits more than that, it will not be accepted and an error is returned before it reaches this code--so I think adding any further limit would be redundant.  Still going with struct{} instead of true though.